[CPLP-1475] - Digital-twin-search-logic

smartdeepp opened the pull request.

manojava-gk commented on cx-portal/src/components/pages/DigitalTwins/TwinTable.tsx:
```diff
@@ -73,6 +98,8 @@ const TwinTable = ({ onTwinSelect }: TwinTableProps) => {
         toolbarVariant="ultimate"
         toolbar={{
           onSearch: onSearch,
+          searchExpr: searchValue,
```
Can make use of PageLoadingTable component here.

manojava-gk replied:
you can refer InviteBusiness section

smartdeepp commented:
Hello @manojava-gk , thanks for the suggestion, but it seems like PageLoadingTable takes totally different components and flow to make it operational,  i had a check on the PageLoadingTable, it wont be compatible with the flow we used in Digital Twin. Can you guide me through this?

manojava-gk commented:
> Hello @manojava-gk , thanks for the suggestion, but it seems like PageLoadingTable takes totally different components and flow to make it operational, i had a check on the PageLoadingTable, it wont be compatible with the flow we used in Digital Twin. Can you guide me through this if i am missing anything here?

Hi @smartdeepp thanks for this information. Can we connect over a call in CatenaX Teams? Kindly let me know your email for the same

smartdeepp commented:
> > Hello @manojava-gk , thanks for the suggestion, but it seems like PageLoadingTable takes totally different components and flow to make it operational, i had a check on the PageLoadingTable, it wont be compatible with the flow we used in Digital Twin. Can you guide me through this if i am missing anything here?
> 
> Hi @smartdeepp thanks for this information. Can we connect over a call in CatenaX Teams? Kindly let me know your email for the same

Hello @manojava-gk , sure. it's deep.patel@smartsensesolutions.com .

manojava-gk commented:
> > > Hello @manojava-gk , thanks for the suggestion, but it seems like PageLoadingTable takes totally different components and flow to make it operational, i had a check on the PageLoadingTable, it wont be compatible with the flow we used in Digital Twin. Can you guide me through this if i am missing anything here?
> > 
> > 
> > Hi @smartdeepp thanks for this information. Can we connect over a call in CatenaX Teams? Kindly let me know your email for the same
> 
> Hello @manojava-gk , sure. it's [deep.patel@smartsensesolutions.com](mailto:deep.patel@smartsensesolutions.com) .

Lets connect in teams

JaneMcBrain reviewed (commented):
I reviewed your PR as requested, but I could not checkout your branch, since it is not published inisde the product-portal-frontend. Can you explain where to find it? Would be great to test the UI.

smartdeepp commented:
> I reviewed your PR as requested, but I could not checkout your branch, since it is not published inisde the product-portal-frontend. Can you explain where to find it? Would be great to test the UI.

Can you please check with this branch name `smartdeepp:CPLP-1475-Digital-twin-search-logic`

JaneMcBrain commented:
> > I reviewed your PR as requested, but I could not checkout your branch, since it is not published inisde the product-portal-frontend. Can you explain where to find it? Would be great to test the UI.
> 
> Can you please check with this branch name `smartdeepp:CPLP-1475-Digital-twin-search-logic`

I tried this already, but it is not working. I also deleted my local version and cloned the Portal code again, but still your branch is missing. When you go to https://github.com/catenax-ng/product-portal-frontend/branches I can also not see your branch there... Maybe it has something to do with this?

smartdeepp commented:
> > > I reviewed your PR as requested, but I could not checkout your branch, since it is not published inisde the product-portal-frontend. Can you explain where to find it? Would be great to test the UI.
> > 
> > 
> > Can you please check with this branch name `smartdeepp:CPLP-1475-Digital-twin-search-logic`
> 
> I tried this already, but it is not working. I also deleted my local version and cloned the Portal code again, but still your branch is missing. When you go to https://github.com/catenax-ng/product-portal-frontend/branches I can also not see your branch there... Maybe it has something to do with this?

i have added a new commit too, with the current change you gave.

JaneMcBrain commented:
> > > > I reviewed your PR as requested, but I could not checkout your branch, since it is not published inisde the product-portal-frontend. Can you explain where to find it? Would be great to test the UI.
> > > 
> > > 
> > > Can you please check with this branch name `smartdeepp:CPLP-1475-Digital-twin-search-logic`
> > 
> > 
> > I tried this already, but it is not working. I also deleted my local version and cloned the Portal code again, but still your branch is missing. When you go to https://github.com/catenax-ng/product-portal-frontend/branches I can also not see your branch there... Maybe it has something to do with this?
> 
> i have added a new commit too, with the current change you gave.

Cool! I know now, why I can not checkout to your branch. You forked the project, so it is a different Repo... I don't know if the workflow is good like that, because then the reviewers has to checkout your repo as well. Maybe for the future it is better to just make a branch in the original repository?

smartdeepp commented:
> > > > > I reviewed your PR as requested, but I could not checkout your branch, since it is not published inisde the product-portal-frontend. Can you explain where to find it? Would be great to test the UI.
> > > > 
> > > > 
> > > > Can you please check with this branch name `smartdeepp:CPLP-1475-Digital-twin-search-logic`
> > > 
> > > 
> > > I tried this already, but it is not working. I also deleted my local version and cloned the Portal code again, but still your branch is missing. When you go to https://github.com/catenax-ng/product-portal-frontend/branches I can also not see your branch there... Maybe it has something to do with this?
> > 
> > 
> > i have added a new commit too, with the current change you gave.
> 
> Cool! I know now, why I can not checkout to your branch. You forked the project, so it is a different Repo... I don't know if the workflow is good like that, because then the reviewers has to checkout your repo as well. Maybe for the future it is better to just make a branch in the original repository?

Yes, you are correct you need to checkout from my repo. Additionally this work flow was given to us by @jjeroch herself. Can you please share your thoughts here @jjeroch

JaneMcBrain commented:
> > > > > > I reviewed your PR as requested, but I could not checkout your branch, since it is not published inisde the product-portal-frontend. Can you explain where to find it? Would be great to test the UI.
> > > > > 
> > > > > 
> > > > > Can you please check with this branch name `smartdeepp:CPLP-1475-Digital-twin-search-logic`
> > > > 
> > > > 
> > > > I tried this already, but it is not working. I also deleted my local version and cloned the Portal code again, but still your branch is missing. When you go to https://github.com/catenax-ng/product-portal-frontend/branches I can also not see your branch there... Maybe it has something to do with this?
> > > 
> > > 
> > > i have added a new commit too, with the current change you gave.
> > 
> > 
> > Cool! I know now, why I can not checkout to your branch. You forked the project, so it is a different Repo... I don't know if the workflow is good like that, because then the reviewers has to checkout your repo as well. Maybe for the future it is better to just make a branch in the original repository?
> 
> Yes, you are correct you need to checkout from my repo. Additionally this work flow was given to us by @jjeroch herself. Can you please share your thoughts here @jjeroch

Ok, so I tested your branch now. Looks good to me. There is just one thing (and it could be possible, that it is a problem of the Table-Component, and not your changes) - when you enter a value in the search field manually it already tries to search after the 3rd or 4th letter and then you can not continue entering, because the input is not active anymore... But the same happens in the semantic model page as well. So I assume this is a different thing.
